quarkchain/evm/tests/test_specials.py

```diff
@@ -99,3 +106,59 @@ def test_proc_balance_mnt(self):
             self.assertEqual(result, 1)
             self.assertEqual(gas_remained, 500 - 400)
             self.assertEqual(ret_int, bal)
+
+    def test_proc_transfer_mnt(self):
+        sender = b"\x00" * 19 + b"\x34"
+        precompiled = decode_hex(b"000000000000000000000000000000514b430002")
+        token_id = 1234567
+        token_id_bytes = token_id.to_bytes(32, byteorder="big")
+        state = State()
+
+        self.__mint(
+            state,
+            sender,
+            token_id_encode("QKC").to_bytes(32, byteorder="big"),
+            encode_int32(30000),
+        )
+        self.__mint(state, sender, token_id_bytes, encode_int32(100))
+        balance = state.get_balance(sender, token_id)
+        self.assertEqual(balance, 100)
+        balance = state.get_balance(sender, token_id_encode("QKC"))
+        self.assertEqual(balance, 30000)
+
+        new_addr = b"\x01" * 20
+        testcases = [
+            # Format: (index, description, to, value, gas, expected ret, post check)
+            (1, "gas not enough", sender, 0, 699, (0, 0, []), None),
+            (2, "no value transfer with min gas", sender, 0, 700, (1, 0, []), None),
+            (3, "value transfer needs more gas", sender, 1, 9699, (0, 0, []), None),
+            # Should have stipend gas left
+            (4, "value transfer needs more gas", sender, 1, 9700, (1, 2300, []), None),
```

Comment:
@qizhou: this is how stipend would work, minimum gas cost met (GCALL 700 + GVALUETRANSFER 9000 = 9700), while still having 2300 left (GSTIPEND), which essentially lowers the cost of GVALUETRANSFER from 9000 to 6700

Reply:
Great!  That is much clearer!